src/rgw/rgw_rest_role.cc

```diff
@@ -131,11 +131,16 @@ void RGWCreateRole::execute(optional_yield y)
   if (op_ret < 0) {
     return;
   }
+  std::string user_tenant = s->user->get_tenant();
   std::unique_ptr<rgw::sal::RGWRole> role = store->get_role(role_name,
-							    s->user->get_tenant(),
+							    user_tenant,
 							    role_path,
 							    trust_policy,
 							    max_session_duration);
+  if (!user_tenant.empty() && role->get_tenant() != user_tenant) {
+    op_ret = -EACCES;
```

Comment:
how about -EINVAL, instead of -EACCES, and log an error message saying that the tenant provided in the role name does not match with the tenant of the user creating the role or something similar.

Reply:
I was hesitant about -EINVAL or -EACCES, now -EINVAL looks more universal, log message has been added.

Reply:
@BryceCao : One last comment, please modify the commit message, from delete tenant-parse to something appropriate.